pkg/odo/cli/config/set.go

```diff
@@ -66,13 +70,20 @@ func (o *SetOptions) Complete(name string, cmd *cobra.Command, args []string) (e
 	if err != nil {
 		return err
 	}
-	o.lci = cfg
+	o.LocalConfigInfo = cfg
+	if o.now {
+		o.EnablePushConfig()
```

Comment:
Why if someone changes the code, and then run `odo config set memory 2G --now`?
In this case, we should push both, code and config.
If ``odo config set memory 2G --now` is run without prior code changes, it should just push config.

`odo push` is already smart enough to know what type of push is needed, we need to make use of that logic here.